Common/TableProducer/trackToCollisionAssociator.cxx

```diff
@@ -0,0 +1,171 @@
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file trackToCollisionAssociator.cxx
+/// \brief Associates tracks to collisions considering ambiguities
+/// \author Jan Fiete Grosse-Oetringhaus <jan.fiete.grosse-oetringhaus@cern.ch>, CERN
+/// \author Fabrizio Grosa <fgrosa@cern.ch>, CERN
+/// \author Mattia Faggin <mfaggin@cern.ch>, University and INFN Padova
+
+#include "CommonConstants/LHCConstants.h"
+#include "Framework/AnalysisDataModel.h"
+#include "Framework/AnalysisTask.h"
+#include "Framework/ASoAHelpers.h"
+#include "Framework/runDataProcessing.h"
+#include "Common/DataModel/CollisionAssociation.h"
+#include "Common/DataModel/TrackSelectionTables.h"
+
+using namespace o2;
+using namespace o2::framework;
+using namespace o2::framework::expressions;
+using namespace o2::aod;
+
+struct TrackToCollisionAssociation {
+  Produces<TrackAssoc> association;
+
+  Configurable<float> nSigmaForTimeCompat{"nSigmaForTimeCompat", 4.f, "number of sigmas for time compatibility"};
+  Configurable<float> timeMargin{"timeMargin", 0.f, "time margin in ns added to uncertainty because of uncalibrated TPC"};
+  Configurable<bool> applyTrackSelForRun2{"applyMinimalTrackSelForRun2", false, "flag to apply minimal track selection for Run 2 in case of standard association"};
+  Configurable<bool> applyIsGlobalTrack{"applyIsGlobalTrackWoDCA", true, "flag to apply global track w/o DCA selection"};
+  Configurable<bool> usePVAssociation{"usePVAssociation", true, "if the track is a PV contributor, use the collision time for it"};
+  Configurable<bool> includeUnassigned{"includeUnassigned", false, "consider also tracks which are not assigned to any collision"};
+  Configurable<bool> debug{"debug", false, "fill a table with flag to keep track of kind of track (PV contributor or ambiguous)"};
+
+  Filter trackFilter = (applyIsGlobalTrack == false) || requireGlobalTrackWoDCAInFilter();
+  using TracksWithSel = soa::Join<Tracks, TracksExtra, TrackSelection>;
+  using TracksWithSelFilter = soa::Filtered<TracksWithSel>;
+
+  void init(InitContext const&)
+  {
+    std::array<int, 2> doProcess = {doprocessAssocWithTime, doprocessStandardAssoc};
+    int doProcessSum = std::accumulate(doProcess.begin(), doProcess.end(), 0);
+    if (doProcessSum != 1) {
+      LOGP(fatal, "Exactly one process function should be enabled! Exit");
+    }
+
+    if (applyIsGlobalTrack && applyTrackSelForRun2) {
+      LOGP(fatal, "You cannot apply simultaneously Run 2 track selections and isGlobalWoDCA! Exit");
```

Comment:
Maybe `isGlobalWoDCA` can be replaced with the correct name appearing in the code?